web-app/src/main/webapp/script.js

```diff
@@ -93,24 +94,42 @@ function showMessageOnInfoWindow(message, position, map, infoWindow) {
   infoWindow.open(map);
 }
 
-function showReportForm() {
+function showReportForm(map, geocoder) {
   document.getElementById('form-container').style.display = 'block';
   const homeElements = document.getElementsByClassName('home');
   for (const element of homeElements) {
     element.style.display = 'none';
   }
+
+  geocoder.geocode({ 'location': map.getCenter() }, (results, status) => {
+    if (status === 'OK') {
+      if (results[0]) {
+        document.getElementById('location-input').value = results[0].formatted_address;
```

Comment:
I sense an occasional stringified `"undefined"` showing up in the user interface :)
You may want to handle that, but up to you.